File: 02-semantics.rb
# SEMANTICS!

# Evaluation rules
1 == '1'
1 == '1'.to_i

[1, 2, 3] == [1, 2, 3]
foo = [1, 2, 3]
foo == foo

# Developer friendly!
beatles = ["John", "Paul", "George", "Ringo"]
beatles - ["John"]

beatles == ["Ringo", "George", "Paul", "John"]
(beatles - ["Ringo", "George", "Paul", "John"]) == []

# Boolean

# Ruby is really, really object-oriented
3.class
3.abs
-3.abs
"giraffe".length
a = "giraffe"
a.reverse

File: 02-semantics.js
// SEMANTICS!

// Evaluation rules
1 == '1'
1 == parseInt('1')
1 === '1'

// Array comparisons
// UGH https://stackoverflow.com/questions/7837456/how-to-compare-arrays-in-javascript
[1, 2, 3] == [1, 2, 3]

var foo = [1, 2, 3];
foo == foo
